File: scripts/task_tested_with_compile_script/factorial/c/factorial.c
#include <stdio.h>


int main() {

volatile int result;

for (int i = 0; i < 100000; ++i) {
	result = executeTask(i);
}

return 0;
}

int executeTask(int i){
		return factorial(i);
}

int factorial(int n) {
    return n == 0 ? 1 : n * factorial(n - 1);
}

File: scripts/task_tested_with_compile_script/factorial/perl/factorial.pl

my $r = 1;

sub executeTask {
	my ($i) = @_;
	return factorial( $i);
}

foreach ( $a = 0; $a < 100000; $a + $a + 1 ) {
	$r = executeTask($a);
}

sub factorial
{
   my $number = shift;   # get the argument

   if ( $number <= 1 ) { # base case
      return 1;
   }
   else {                # recursive step
      return $number * factorial( $number - 1 );
   }
}
